sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -188,14 +188,17 @@ static inline std::string GetHTTPMessagePreBody(Azure::Core::Http::Request const
 
 static void CleanupThread()
```

Comment:
What if we don't invoke this one in destructor, but instead, write a `Finalize()` function, and invoke it from `atexit()`. This way, when you do all that work, it is not during the statics destruction, when half of other statics can be destructed already.

Reply:
Static destructors are usually implemented by registering cleanup functions with `atexit`, so I'd expect this to make little difference.

Reply:
> What if we don't invoke this one in destructor

It's not invoked from any destructor.  When the libcurl thread pool moves a connection back to the pool, we spawn a new thread that runs this static method and we detach the thread from main thread.

The method is a loop that executes every amount of time and check if there are old connections to be removed from the pool.

If the pool is empty, the method just returns and the thread is joined to main thread to be ended.  Then, a new clean thread would be spawn if a connection is again moved back to the pool.

We don't want to clean on application exit because that could lead to have an application with hundreds of useless connections in the pool, just wasting memory.
We don't want to clean connections upon getting a new connection because that might delay an SDK client from performing a request.
We don't want to clean connections when a connection is back to the pool because if a program creates one million connections, use them all and then all connections are moved back to the pool, the clean routine would not clean anything on that moment and the connections would stay in the pool forever.